### PR TITLE
Add visibility change listener for performance telemetry

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Adds counters to `PerformanceEvent`.
- Adds a method to increment counters to `PerformanceEvent`. 
- Adds a `visibilityChangeCount` to performance data to count Page Visibility changes during the AcquireTokenSilent, AcquireTokenByCode, SsoSilent calls. 
- Tests included.

### CODE REVIEW

**Comment by konstantin-msft** [comment]
Copy of https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5488. All code review comments are addressed.

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 1]
```diff
@@ -415,6 +419,20 @@ export abstract class ClientApplication {
...
 
+    private trackPageVisibilityWithMeasurement():void {
+        const measurement = this.ssoSilentMeasurement || this.acquireTokenByCodeAsyncMeasurement;
+        if(!measurement) {
+            return;
+        }
+
+        this.logger.info("Perf: Visibility change detected in ", measurement.event.name);
+        measurement.addStaticFields({
+            visibilityChange: true,
```
Comment (peterzenz) [inline lib/msal-browser/src/app/ClientApplication.ts]: I think the visibility change will be really diagnostic but as I'm looking at it, does anyone have any thoughts on whether we should make this a count vs a flag?  For instance, in the course of this call, the visibility changed 3x.

Comment (tnorling) [reply lib/msal-browser/src/app/ClientApplication.ts]: I don't have a strong opinion but my thoughts are "why not"

Comment (konstantin-msft) [reply lib/msal-browser/src/app/ClientApplication.ts]: I think we can. Do you want me to make a change in this pr or make a follow-up?

Comment (peterzenz) [reply lib/msal-browser/src/app/ClientApplication.ts]: I'm thinking of how I use the browser and I often ctrl+click on links which tends to open them in a "hidden" tab.

Let's change it to a counter.  I'm ok with either going in this PR or the next one.  If you want to close this one out as is that's OK with me.

**Review by peterzenz — APPROVED** [review]

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 1 above)
Comment (sameerag) [reply lib/msal-browser/src/app/ClientApplication.ts]: I am leaning towards the counter, which gives a more accurate status.

Comment (konstantin-msft) [reply lib/msal-browser/src/app/ClientApplication.ts]: Done.

**Review by hectormmg — APPROVED** [review]
LGTM!

**Pull request merged**